### PR TITLE
Add strong-names to .NET assemblies.

### PR DESCRIPTION
**Pull request by mpiroc**
Based on #204 and #205. Compare with #205 (branch `pirocchi/publish-metadata` ) to view only new changes.

This change fetches a `.snk` file from secrets manager prior to build. This key is used during build to add strong names to each .NET assembly.

Your CI environment should set the `DOTNET_STRONG_NAME_ROLE_ARN` and `DOTNET_STRONG_NAME_SECRET_ID` environment variables prior to build.

### CODE REVIEW

File: fetch-dotnet-snk.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+set -euo pipefail
+
+# This script retrieves the .snk file needed
+# to create strong names for .NET assemblies.
+echo "Retrieving SNK..."
+
+sudo apt install jq -y
+
+ROLE=$(aws sts assume-role --region us-east-2 --role-arn ${DOTNET_STRONG_NAME_ROLE_ARN:-} --role-session-name "jsii-dotnet-snk")
```
Comment (costleya) [inline fetch-dotnet-snk.sh]: pass in region?

File: fetch-dotnet-snk.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+export AWS_ACCESS_KEY_ID=$(echo $ROLE | jq -r .Credentials.AccessKeyId)
+export AWS_SECRET_ACCESS_KEY=$(echo $ROLE | jq -r .Credentials.SecretAccessKey)
+export AWS_SESSION_TOKEN=$(echo $ROLE | jq .Credentials.SessionToken)
+
+SNK_SECRET=$(aws secretsmanager get-secret-value --region us-east-2 --secret-id ${DOTNET_STRONG_NAME_SECRET_ID:-})
```
Comment (costleya) [inline fetch-dotnet-snk.sh]: pass in region?

File: fetch-dotnet-snk.sh (hunk 1 above)
Comment (mpiroc) [reply fetch-dotnet-snk.sh]: ✔️

File: fetch-dotnet-snk.sh (hunk 2 above)
Comment (mpiroc) [reply fetch-dotnet-snk.sh]: ✔️

**Pull request merged**